### PR TITLE
LPS-192658 Properly index Date array (when they come as String array) and refactor

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPS-192658](https://liferay.atlassian.net/browse/LPS-192658)

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 1]
```diff
@@ -665,6 +666,18 @@ else if (value instanceof Number[]) {
...
+
+			if (type.equals(DDMFormFieldTypeConstants.DATE) ||
+				type.equals(DDMFormFieldTypeConstants.DATE_TIME)) {
+
+				Date[] dateValues = _getDateValues(valuesString, type);
+
+				if (dateValues.length > 0) {
+					document.addDate(name.concat("_date"), dateValues);
+				}
+			}
```
Comment (murilostodolni) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:679]: Can we put this if inside if else below? Something like this:

```java
if (type.equals(DDMFormFieldTypeConstants.DATE) ||
	type.equals(DDMFormFieldTypeConstants.DATE_TIME)) {

	Date[] dateValues = _getDateValues(valuesString, type);

	if (dateValues.length > 0) {
		document.addDate(name.concat("_date"), dateValues);
	}
} else if (indexType.equals("keyword")) {
	document.addKeywordSortable(name, valuesString);
}
else {
	document.addTextSortable(name, valuesString);
}
```

**Review by murilostodolni — COMMENTED** [review]
Hey @devpedrotavares 

Good job! I have just one question.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (murilostodolni) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:679]: Thanks for the explanation @jorgediaz-lr

**Review by murilostodolni — APPROVED** [review]

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Review by carolmariaabb — APPROVED** [review]
LGTM 🎉

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by devpedrotavares** [comment]
ci:forward

**Comment by devpedrotavares** [comment]
ci:forward:force

**Pull request closed without merging**